src/libs/sfdk/dockercontainer.cpp

```diff
@@ -0,0 +1,337 @@
+    arguments.append(CGROUP_MOUNT);
+    arguments.append(TMPFS);
+    arguments.append(TMPFS1);
+    arguments.append(TMPFS);
+    arguments.append(TMPFS2);
+    arguments.append(TMPFS);
+    arguments.append(TMPFS3);
+
+    for (auto &ports : qAsConst(m_reservedPortMappings)) {
+        arguments.append(PUBLISH_PORT);
+        arguments.append(QString::number(ports.first) + ":" + QString::number(ports.second));
+    }
+
+    for (auto &paths : qAsConst(m_sharedPaths)) {
+        arguments.append(VOLUME);
+        arguments.append(paths.first + ":" + paths.second);
+    }
+
+    arguments.append(q->name());
+
+    auto runner = std::make_unique<ProcessRunner>(DockerManager::dockerPath(), arguments);
+    QObject::connect(runner.get(), &ProcessRunner::done, q, [q, functor, process = runner->process()](bool ok) {
+        if (ok)
+            q->m_containerId = QString::fromLocal8Bit(process->readAllStandardOutput().trimmed());
+        functor(ok);
+    });
+    SdkPrivate::commandQueue()->enqueue(std::move(runner));
+}
+
+void DockerContainerPrivate::stop(const QObject *context, const Functor<bool> &functor)
+{
+    Q_Q(DockerContainer);
+    Q_ASSERT(context);
+    Q_ASSERT(functor);
+
+    const QPointer<const QObject> context_{context};
+
+    auto enqueue = [=](const QStringList &arguments, CommandQueue::BatchId batch) {
+           auto runner = std::make_unique<ProcessRunner>(DockerManager::dockerPath(), arguments);
+           QObject::connect(runner.get(), &ProcessRunner::failure, Sdk::instance(), [=]() {
+               SdkPrivate::commandQueue()->cancelBatch(batch);
+               callIf(context_, functor, false);
+           });
+           ProcessRunner *runner_ = runner.get();
+           SdkPrivate::commandQueue()->enqueue(std::move(runner));
+           return runner_;
+       };
+
+    const CommandQueue::BatchId batch = SdkPrivate::commandQueue()->beginBatch();
+
+    QStringList stopArguments;
+    stopArguments.append(STOP);
+    stopArguments.append(q->m_containerId);
+
+    enqueue(stopArguments, batch);
+
+    QStringList commitArguments;
+    commitArguments.append(COMMIT);
+    commitArguments.append(q->m_containerId);
+    commitArguments.append(q->name());
+
+    enqueue(stopArguments, batch);
+
+    SdkPrivate::commandQueue()->enqueueCheckPoint(context, [=]() { functor(true); });
+    SdkPrivate::commandQueue()->endBatch();
+}
+
+void DockerContainerPrivate::probe(const QObject *context,
+                                   const Functor<VirtualMachinePrivate::BasicState, bool> &functor) const
+{
+    Q_Q(const DockerContainer);
+    Q_ASSERT(context);
+    Q_ASSERT(functor);
+
+    DockerManager::fetchRegisteredVirtualMachines(q, [=](const QStringList &registeredVms, bool ok) {
+        VirtualMachinePrivate::BasicState state;
+        if (!ok) {
+            functor(state, false);
+        } else if (!registeredVms.contains(q->name())) {
+            functor(state, true);
+        } else {
```

Comment:
Looking at it now, it would be better to explicitly return from the first two blocks and omit the else block, reducing indentation and possibly adding some blank lines  e.g. after arguments list building.

Reply:
fixed